src/routes/SheltersPege/SheltersPege.js

```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
     return filters;
   }
 
+  updateState(shelter) {
+    this.setState({ shelterName: shelter[0].ShelterName })
+    this.setState({ address: shelter[0].Address.substring(0, shelter[0].Address.indexOf(',')) })
+    this.setState({ city: shelter[0].Address.substring(shelter[0].Address.indexOf(',') + 2, shelter[0].Address.lastIndexOf(',')) })
+    this.setState({ state: shelter[0].Address.substring(shelter[0].Address.lastIndexOf(',') + 2, shelter[0].Address.lastIndexOf(',') + 4) })
+    this.setState({ zip: parseInt(shelter[0].Address.substring(shelter[0].Address.length - 5)) })
+    this.setState({ email: shelter[0].EmailAddress })
+    this.setState({ phoneNumber: parseInt(shelter[0].PhoneNumber) })
+    this.setState({ website: shelter[0].Website })
+  }
+
+  onClickEdit() {
+    const { shelter } = this.state;
+    if (this.state.mode === "view") {
+      this.updateState(shelter)
+      this.setState({ mode: "edit" })
+    } else {
+      this.setState({ mode: "view" })
+    }
+  }
+
+  onClickDelete() {
+    if (window.confirm('Are you sure you want to delete this shelter?')) {
+      const { shelterID } = this.state;
+      SheltersService.deleteShelter(shelterID)
+        .then(() => {
+          localStorage.removeItem('token')
+          window.open("/");
+        })
+        .catch(res => {
+          window.alert(res.error);
+        });
+    }
+  };
+
+  onClickUpdate() {
+    const {
+      shelterID,
+      shelterName,
+      address,
+      city,
+      state,
+      zip,
+      email,
+      phoneNumber,
+      website
+    } = this.state;
+
+    let data = {
+      "shelterID": parseInt(shelterID),
+      "shelterName": shelterName,
+      "address": address + ", " + city + ", " + state + " " + zip,
+      "emailAddress": email,
+      "phoneNumber": parseInt(phoneNumber),
```

Comment:
Is `parseInt()` necessary? I thought the DB was now storing this value as a string

Reply:
good catch.

Reply:
Looks like this wasn't updated